code/backend/FakeMyResume/Controllers/UserController.cs

```diff
@@ -13,11 +14,17 @@ public class UserController : Controller
             _userService = userService;
         }
         [HttpGet("current")]
-        public IActionResult GetUser()
+        public async Task<IActionResult> GetUserAsync()
         {
             var accountId = User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
             var email = User.Identity.Name;
             var userInfo = _userService.GetUserByUserName(email);
+            if (User.Identity.IsAuthenticated && !string.IsNullOrEmpty(User.Identity.Name) && userInfo == null)
+            {//the user was not in the system but is a valid user
+                FakeMyResume.Models.Data.User user = new FakeMyResume.Models.Data.User();
+                user.UserName = User.Identity.Name;
+                userInfo = await _userService.CreateUserAsync(user);
```

Comment:
Saving a user that has not interacted with the system is unnecessary, we only need to save users that submit a resume. This whole /current flow needs to go away  or change.